### DISCUSSION TITLE
Playlet Backend

### DISCUSSION THREAD

**Discussion by rammjet** [iBicha/playlet#520] [General]
Removing my invidious link in the Playlet app interface wasn't enough.  I had to login to the server at http://myip:8888 and logout in my profile.

Now, how do I get subscriptions and playlists into the Playlet backend?

**Comment by iBicha** [comment]
When you change the Invidious instance in the settings, your current profile remains on the previous instance. Profiles belong to their respective instances, they are not portable.

The playlet backend does not support accounts yet. Invidious instances have real servers and databases to store users, subscriptions, and so on. This is not possible to do on a Roku device.

If Playlet supports accounts one day, it will be regular YouTube accounts. But right now, we're not there yet.

**Comment by rammjet** [comment]
So, when you say that we will no longer need an invidious instance, we really need one to maintain the account for subscriptions and playlists even if it is not connecting to YouTube?

**Reply by iBicha** [reply]
That's right. There are three things that are tied to accounts: Subscriptions, Playlists (saved playlists in an Invidious profile) and Watch history. These can't be realistically stored on device with only **32KB of storage** (yes, that's the space available to Roku channels) - and also a Roku device is not powerful enough to query the latest videos from the subscription list by itself (without a server/database)

**Comment by rammjet** [comment]
I'm not criticizing, just trying to understand.  The info below led me to believe that I would no longer need invidious:


    Built-in Playlet backend, which supports channels, playlists, and search
        This backend is experimental, and has a few missing features.
        To use it, set the Invidious instance to an empty string.

**Reply by iBicha** [reply]
No worries, happy to clarify: you can open channels, playlists, and search.
I also phrase it differently in the announcement "This allows you to access YouTube content without the need for an Invidious instance." So you can access content, and this is especially useful when all instances are down/api blocked.

**Reply by QuietImCoding** [reply]
Hey, huge fan of Playlet here, trying to find ways to help support the app, it seems like Invidious is basically becoming unsupported nowadays, so would it be helpful to create some sort of very simple API that allows storing Subscriptions, Playlists and Watch History? 

It seems like those things could be stored in a very small db external to the Playlet app itself, maybe a tiny API that exposes just the invidious Account endpoints?

**Reply by iBicha** [reply]
> Hey, huge fan of Playlet here, trying to find ways to help support the app, it seems like Invidious is basically becoming unsupported nowadays, so would it be helpful to create some sort of very simple API that allows storing Subscriptions, Playlists and Watch History?
> 
> It seems like those things could be stored in a very small db external to the Playlet app itself, maybe a tiny API that exposes just the invidious Account endpoints?

Thanks for your suggestion, Invidious is not becoming unsupported, it's still working well, it's the lack of public instances hosted by volunteers that disappeared.

So if someone wants to go through the effort to create an API that is compatible with Invidious account endpoints, might as well host Invidious, or a fork of Invidious where only the account stuff is exposed.
Don't forget that it's not going to be just "very small db" you will still need to query channels at a regular intervals to cache a sort subscription videos, and also fetch the information of playlists/videos in order to serve user playlists. As I said, might as well host Invidious.

I will be working on supporting YouTube accounts login in the future, but I need to make significant changes to the app, so it might take a few months (that if I find the time to do this) - but it is coming

**Comment by freeagede** [comment]
Hi now one Week its not working any more. I try every Day

**Comment by QuietImCoding** [comment]
Oh interesting! Thank you for the thoughtful response, I guess I mistakenly conflated the public instances going down with the whole app becoming deprecated. I will try to host my own instance then!
